micronota/parsers/cmscan.py

```diff
@@ -0,0 +1,413 @@
+   'SEQUENCE_ACCESSION':      {'position':  3, 'width':  9, 'textalignment': 'l', 'caption': 'accession'},
+   'TYPE_OF_MODEL':           {'position':  4, 'width':  3, 'textalignment': 'r', 'caption': 'mdl'},
+   'MODEL_START_POSITION':    {'position':  5, 'width':  8, 'textalignment': 'r', 'caption': 'mdl from'},
+   'MODEL_END_POSITION':      {'position':  6, 'width':  8, 'textalignment': 'r', 'caption': 'mdl to'},
+   'SEQUENCE_START_POSITION': {'position':  7, 'width':  8, 'textalignment': 'r', 'caption': 'seq from'},
+   'SEQUENCE_END_POSITION':   {'position':  8, 'width':  8, 'textalignment': 'r', 'caption': 'seq to'},
+   'STRAND':                  {'position':  9, 'width':  6, 'textalignment': 'r', 'caption': 'strand'},
+   'TRUNCATED':               {'position': 10, 'width':  5, 'textalignment': 'r', 'caption': 'trunc'},
+   'PASS':                    {'position': 11, 'width':  4, 'textalignment': 'r', 'caption': 'pass'},
+   'GC_CONTENT':              {'position': 12, 'width':  4, 'textalignment': 'r', 'caption': 'gc'},
+   'BIAS':                    {'position': 13, 'width':  5, 'textalignment': 'r', 'caption': 'bias'},
+   'BITSCORE':                {'position': 14, 'width':  6, 'textalignment': 'r', 'caption': 'score'},
+   'EVALUE':                  {'position': 15, 'width':  9, 'textalignment': 'r', 'caption': 'E-value'},
+   'INC':                     {'position': 16, 'width':  3, 'textalignment': 'l', 'caption': 'inc'},
+   'DESCRIPTION':             {'position': 17, 'width': 21, 'textalignment': 'l', 'caption': 'description of target'}
+}
+_orderedKeys = sorted(_COLUMNS, key=lambda x: _COLUMNS[x]['position'])
+
+
+class CmscanFormatError(FileFormatError):
+    pass
+
+
+def _construct(record, constructor=None, **kwargs):
+    if constructor is None:
+        constructor = IntervalMetadata
+    if constructor is IntervalMetadata:
+        return IntervalMetadata(features=record)
+
+
+@cmscan.sniffer()
+def _cmscan_sniffer(fh):
+    # check the 1st real line is a valid ID line
+    if _too_many_blanks(fh, 5):
+        return False, {}
+
+    try:
+        line = next(_line_generator(fh, skip_blanks=True, strip=False))
+    except StopIteration:
+        return False, {}
+
+    if line.startswith('#target name'):
+        return True, {}
+    else:
+        return False, {}
+
+
+@cmscan.reader(None)
+def _cmscan_to_generator(fh, constructor=None, **kwargs):
+    for record in _parse_records(fh):
+        yield _construct(record, constructor, **kwargs)
+
+
+@cmscan.reader(IntervalMetadata)
+def _cmscan_to_metadata(fh, rec_num=1, **kwargs):
+    record = _get_nth_record(_parse_records(fh), rec_num)
+    return _construct(record, IntervalMetadata, **kwargs)
+
+
+def _parse_records(fh):
+    """ parses an tabular output file, generated by either cmsearch or cmscan
+    of the Infernal package, to a collection of IntervalMetadata objects
+
+    Parameters
+    ----------
+    fh : a file handle to the file that should be parsed
```

Comment:
We typically use [numpy docs](https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt).

For instance, you'll want to write the docs for the `fh` as follows
```
Parameters
----------
fh : str
   A file handle to the file that should be parsed.
```